app/controllers/TransactionalController.scala

```diff
@@ -84,5 +84,28 @@ trait TransactionalController extends BaseController {
         case _ => NoContent
       }
   }
+
+  def fetchSicCodesByTransactionID(txId: String) = Action.async {
+    implicit request =>
+      service.checkIfCompIncorporated(txId).flatMap {
+        case Some(crn) => service.fetchSicCodes(crn, usePublicAPI = true) flatMap {
```

Comment:
also surely this logic should be the same as the fetchSicCodesByCRN. if incorp and crn exists, call fetchSicCodes once as this is the most up to date siccodes not stale data